### PR TITLE
Pilot 1507: remove /dataset/verify API

### PR DESCRIPTION
**Pull request by QXgu**
## Summary

Pilot 1507: remove /dataset/verify API

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1508

## Type of Change

Please delete options that are not relevant.

- [ ] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [ ] Yes
- [ ] No

## Test Directions

(Additional instructions for how to run tests or validate functionality if not covered by unit tests)

### CODE REVIEW

File: app/routers/v1/api_dataset_restful.py [hunk 1]
```diff
@@ -106,67 +98,6 @@ async def create_dataset(
...
-            await run_in_threadpool(shutil.rmtree, TEMP_FOLDER + dataset.code)
-        except Exception:
-            res.code = EAPIResponseCode.internal_error
-            res.result = 'failed to remove temp bids folder'
-            return res.json_response()
-
-        res.result = json.loads(result.stdout)
-        return res.json_response()
 
     @router.post('/v1/dataset/verify/pre', tags=[_API_TAG], summary='pre verify a bids dataset.')
```
Comment (afmam) [inline app/routers/v1/api_dataset_restful.py:102]: Is this `/v1/dataset/verify/pre` still in use?

File: app/routers/v1/api_dataset_restful.py [hunk 2]
```diff
@@ -106,67 +98,6 @@ async def create_dataset(
             res.result = created
             return res.json_response()
 
-    @router.post('/v1/dataset/verify', tags=[_API_TAG], summary='verify a bids dataset.')
-    @catch_internal(_API_NAMESPACE)
-    async def verify_dataset(
-        self, request_payload: DatasetVerifyForm, dataset_crud: DatasetCRUD = Depends(get_dataset_crud)
```
Comment (afmam) [inline app/routers/v1/api_dataset_restful.py:112]: If `/v1/dataset/verify/pre` is not in use, we can remove the DatasetVerifyForm model as well from https://github.com/PilotDataPlatform/dataset/blob/develop/app/schemas/reqres_dataset.py

**Review by afmam — CHANGES_REQUESTED** [review]
Made 2 comments, also tests are failing.

File: app/routers/v1/api_dataset_restful.py (hunk 2 above)
Comment (QXgu) [reply app/routers/v1/api_dataset_restful.py:112]: /v1/dataset/verify/pre is still used in bff service for bids validation

File: app/routers/v1/api_dataset_restful.py (hunk 1 above)
Comment (QXgu) [reply app/routers/v1/api_dataset_restful.py:101]: @afmam /v1/dataset/verify/pre this API is still in use. I can not deprecate this one

File: app/routers/v1/api_dataset_restful.py (hunk 2 above)
Comment (afmam) [reply app/routers/v1/api_dataset_restful.py:112]: @QXgu thanks for confirming

File: app/routers/v1/api_dataset_restful.py (hunk 1 above)
Comment (afmam) [reply app/routers/v1/api_dataset_restful.py:101]: @QXgu thanks for confirming

**Review by afmam — APPROVED** [review]

**Pull request merged**